apiv6/kentikapi/httputil/httputil_test.go

```diff
@@ -0,0 +1,120 @@
+package httputil
+
+import (
+	"context"
+	"errors"
+	"net"
+	"net/http"
+	"net/url"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// Sequence of calls for single request:
+// 1. retryingClient.Do()
+// 2. retryingClient.retryableRoundTripper.RoundTrip()
+// 3. retryingClient.retryableRoundTripper.retryableClient.Do()
+// 4. retryingClient.retryableRoundTripper.retryableClient.httpClient.Do()
+// 5. retryingClient.retryableRoundTripper.retryableClient.httpClient.httpTransport.RoundTrip()
+
+func TestRetryingClient_Do_ReturnsHTTPTransportError(t *testing.T) {
+	// arrange
+	c := NewRetryingClient(ClientConfig{})
+
+	req, err := http.NewRequestWithContext(context.Background(), http.MethodGet, "https://invalid.url", nil)
+	require.NoError(t, err)
+
+	// act
+	resp, err := c.Do(req)
+
+	// assert
+	t.Logf("Got response: %v, err: %v", resp, err)
+
+	var dnsErr *net.DNSError
+	require.True(t, errors.As(err, &dnsErr))
+	assert.Equal(t, "no such host", dnsErr.Err)
```

Comment:
Not sure if error str comparison is reliable

Reply:
It is OK for a test.

Reply:
It is a test and the chances that it will change are very low (it's a constant in stdlib).